client/scala/src/main/scala/ai/verta/blobs/dataset/AtlasHiveDatasetBlob.scala

```diff
@@ -0,0 +1,150 @@
+package ai.verta.blobs.dataset
+
+import net.liftweb.json._
+import ai.verta.swagger.client.objects._
+import ai.verta.swagger.client.HttpClient
+import ai.verta.client.entities.utils._
+
+import scala.concurrent.{Await, ExecutionContext}
+import scala.util.{Try, Success, Failure}
+import scala.concurrent.duration.Duration
+
+/** Represents a dataset from a query to an Atlas Hive table.
+ */
+class AtlasHiveDatasetBlob(
+  private val atlasQuery: String,
+  private val atlasSourceURI: String,
+  override val numRecords: Option[BigInt] = None,
+  override val executionTimestamp: Option[BigInt] = None,
+  val tags: List[String],
+  val attributes: Map[String, ValueType]
+) extends QueryDatasetBlob {
+  // cannot make this a case class due to constructor conflict.
+
+  override val query = Some(atlasQuery)
+  override val dataSourceURI = Some(atlasSourceURI)
+}
+
+object AtlasHiveDatasetBlob {
```

Comment:
Same feedback on the name. It's ok to only support hive for now, but I'm not sure that should end up being different classes.

Reply:
Fixed. AtlasHiveDatastBlob => AtlasDatasetBlob.